pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -412,10 +412,25 @@ class Timestamp(_Timestamp):
                 )
 
         elif ts_input is _no_input:
-            # User passed keyword arguments.
-            ts_input = datetime(year, month, day, hour or 0,
-                                minute or 0, second or 0,
-                                microsecond or 0)
+            # GH 31200
+            # When year, month or day is not given, we call the datetime
+            # constructor to make sure we get the same error message
+            # since Timestamp inherits datetime
+            datetime_kwargs = {
+                "hour": hour or 0,
+                "minute": minute or 0,
+                "second": second or 0,
+                "microsecond": microsecond or 0
+            }
```

Comment:
can't you just 

```
datetime_kwargs = {
....
year = year,
month=month,
day=day}
```

Reply:
IOW put all of the args in a dict

Reply:
We don't want to put e.g. `'year': None` in the dict because it produces a different error message than what we want.

Reply:
Yes, as @mroeschke pointed out, we can't put arguments `year`, `month` or `day` in `datetime_kwargs` because it products a different error message. Doing that is equivalent to `datetime(year=None, month=None, day=None)` whereas we want it to behave like `datetime()` and so that the error message explicitly tells users they are missing an argument (e.g. year), rather than a confusing `integer is required`